frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -406,37 +405,65 @@ export const removeResourceConnection = (
   });
 };
 
+const safeKill = async (model: K8sKind, obj: K8sResourceKind) => {
+  const resp = await checkAccess({
+    group: model.apiGroup,
+    resource: model.plural,
+    verb: 'delete',
+    name: obj.metadata.name,
+    namespace: obj.metadata.namespace,
+  });
+  if (resp.status.allowed) {
+    return k8sKill(model, obj);
+  }
+  return null;
+};
+
+const deleteWebhooks = (
+  resource: K8sResourceKind,
+  workload: TopologyDataObject<{ isKnativeResource?: boolean }>,
+) => {
+  const isKnativeResource = workload?.data?.isKnativeResource ?? false;
+  const deploymentsAnnotations = resource.metadata?.annotations ?? {};
+  const gitType = detectGitType(deploymentsAnnotations['app.openshift.io/vcs-uri']);
+  const buildConfigs = workload?.resources?.buildConfigs;
+  return buildConfigs?.reduce((requests, bc) => {
+    const triggers = bc.spec?.triggers;
+    const reqs = triggers?.reduce((a, t) => {
+      let obj: K8sResourceKind;
+      const webhookType = t.generic ? 'generic' : gitType;
+      const webhookTypeObj = t.generic || (!isKnativeResource && t[gitType]);
+      if (webhookTypeObj) {
+        obj = {
+          ...resource,
+          metadata: {
+            name:
+              webhookTypeObj.secretReference?.name ??
+              `${resource.metadata.name}-${webhookType}-webhook-secret`,
+            namespace: resource.metadata.namespace,
+          },
+        };
+      }
+      return obj ? [...a, safeKill(SecretModel, obj)] : a;
+    }, []);
+    return [...requests, ...reqs];
```

Comment:
`Uncaught (in promise) TypeError: reqs is not iterable`

If you don't get triggers it throws an exception. You need to guard against spreading `undefined`.

Reply:
fixed!